### PR TITLE
Add react app

### PR DESCRIPTION
**Pull request by starsinmypockets**
Add entire react project here.

### CODE REVIEW

**Comment by adborden** [comment]
@starsinmypockets just a quick gut check, are you sure you want to commit the entire prototype app as-is? I was thinking we would throw away the prototype and start from scratch, selectively pulling pieces in as needed. I'm just seeing a lot of things I would want to remove anyway, so why create bloat in the git history?

For example, USWDS should be pulled in by npm/yarn, not committed in the repo. I also see other files included here which seem like they're part of a sample app and we'd want to remove anyway.

- `app/__fixtures__/Collections.js`
- app/cypress/videos/required-metadata.spec.js.mp4
- app/public/logo192.png
- app/screenshot.png

How would you like to handle it?

**Comment by starsinmypockets** [comment]
@adborden Maybe I misunderstood your initial comments on the repo to mean that you thought that we should include the react development environment as well as the ckan extension. I don't think that this is a bad way to go, but if that is not a requirement for the current work, then I'm happy to revert this to what we had, which was a built prototype of said app in the `public` folder.

**Comment by adborden** [comment]
We chatted about this in our sync, but just wanted to include it here (since I've already written in :slightly_smiling_face:).

@starsinmypockets yeah, sorry for not being clear, there's two things I want, and maybe not part of this story.

I **do** want any JS source files included in this repo (we shouldn't have two separate repos for this extension).
I **don't** want to start with the existing React prototype.

A prototype is an experiment that we learn from and then throw away. We should only be building off of a prototype if there is a really good reason to. Don't forget the prototype was built as part of a 3 day research story, with no tests, by a developer unfamiliar with our team practices and coding conventions, and was not code reviewed. The output of the research story is not the prototype itself, but what we learned from it. The prototype is still useful as a reference and maybe even we want to keep some pieces of it, like the JS build system, but we should be selectively pulling in those pieces and **not** using it as a foundation.

tl;dr
For this story https://github.com/GSA/datagov-ckan-multi/issues/327, I'm really just looking for a "hello world" app. So we could leave the compiled files as-is, and then in the next story (https://github.com/GSA/datagov-ckan-multi/issues/328) we'll add the JS build tooling with USWDS **OR** we could pull in the build tooling now but the React app is an empty "hello world" with none of the CSS or code (essentially just the pieces we like from the prototype needed to build an empty React app).

How does that sound to you?

**Comment by starsinmypockets** [comment]
@adborden Sounds correct.

I'll include the compiled prototype aka the "hello world" app here.

For https://github.com/GSA/datagov-ckan-multi/issues/328 we will start from a clean install of [create-react-app](https://github.com/facebook/create-react-app) which we are using as a base for development.

**Comment by starsinmypockets** [comment]
Will not implement for "hello world" task
